File: codegen/results/v002/qwen3.6_35b_nothink/20260418_170014/python/007_beatles_interview/solution.py
import csv
import json
from datetime import datetime
from io import StringIO

def calculate_age(birthday_str: str) -> int:
    """Calculate age as of July 1, 2025."""
    # Parse the birthday string in MM/DD/YYYY format
    birth_date = datetime.strptime(birthday_str, '%m/%d/%Y')
    # Reference date
    reference_date = datetime(2025, 7, 1)
    
    # Calculate age
    age = reference_date.year - birth_date.year
    # Check if birthday has occurred this year
    if (reference_date.month, reference_date.day) < (birth_date.month, birth_date.day):
        age -= 1
    
    return age

def parse_name(name_str: str) -> tuple:
    """Parse full name into FirstName and LastName.
    
    Handles names with middle names by taking the first and last non-empty parts.
    """
    parts = name_str.strip().split()
    if len(parts) >= 2:
        first_name = parts[0]
        last_name = parts[-1]
        return first_name, last_name
    elif len(parts) == 1:
        return parts[0], ""
    else:
        return "", ""

def get_relative(name_str: str, relationship: str) -> dict:
    """Create a relative dict from name and relationship."""
    if name_str.lower() == 'null' or not name_str.strip():
        return None
    
    first_name, last_name = parse_name(name_str)
    return {
        "FirstName": first_name,
        "LastName": last_name,
        "Relationship": relationship
    }

def main():
    # Read the CSV file
    with open('input/input.csv', 'r') as f:
        reader = csv.DictReader(f)
        rows = list(reader)
    
    result = []
    
    for row in rows:
        # Parse the name
        first_name, last_name = parse_name(row['Name'])
        
        # Calculate age
        age = calculate_age(row['Birthday'])
        
        # Build relatives list
        relatives = []
        
        # Father
        father_rel = get_relative(row['Father'], 'Father')
        if father_rel:
            relatives.append(father_rel)
        
        # Mother
        mother_rel = get_relative(row['Mother'], 'Mother')
        if mother_rel:
            relatives.append(mother_rel)
        
        # Brother
        brother_rel = get_relative(row['Brother'], 'Brother')
        if brother_rel:
            relatives.append(brother_rel)
        
        # Sister
        sister_rel = get_relative(row['Sister'], 'Sister')
        if sister_rel:
            relatives.append(sister_rel)
        
        person = {
            "FirstName": first_name,
            "LastName": last_name,
            "Birthday": row['Birthday'],  # Keep original format
            "Age": age,
            "Relatives": relatives
        }
        
        result.append(person)
    
    # Output JSON array
    print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()

File: codegen/results/v002/qwen3.6_35b_nothink/20260418_170014/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';
import * as path from 'path';

interface Person {
  FirstName: string;
  LastName: string;
  Birthday: string;
  Age: number;
  Relatives: {
    FirstName: string;
    LastName: string;
    Relationship: string;
  }[];
}

interface Relative {
  FirstName: string;
  LastName: string;
  Relationship: string;
}

function parseBirthday(birthdayStr: string): Date {
  const parts = birthdayStr.split('/');
  const month = parseInt(parts[0], 10);
  const day = parseInt(parts[1], 10);
  const year = parseInt(parts[2], 10);
  return new Date(year, month - 1, day);
}

function formatDate(date: Date): string {
  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const day = String(date.getDate()).padStart(2, '0');
  return `${year}-${month}-${day}`;
}

function calculateAge(birthday: string): number {
  const birthDate = parseBirthday(birthday);
  const referenceDate = new Date(2025, 6, 1); // July 1, 2025

  let age = referenceDate.getFullYear() - birthDate.getFullYear();
  const monthDiff = referenceDate.getMonth() - birthDate.getMonth();
  const dayDiff = referenceDate.getDate() - birthDate.getDate();

  if (monthDiff < 0 || (monthDiff === 0 && dayDiff < 0)) {
    age -= 1;
  }

  return age;
}

function parseName(name: string): { firstName: string; lastName: string } {
  const parts = name.trim().split(/\s+/);
  if (parts.length < 2) {
    throw new Error(`Invalid name format: ${name}`);
  }
  const lastName = parts.pop()!;
  const firstName = parts.join(' ');
  return { firstName, lastName };
}

function main() {
  const inputPath = path.join('input', 'input.csv');
  const csvContent = fs.readFileSync(inputPath, 'utf-8');
  const lines = csvContent.trim().split('\n');
  const headers = lines[0].split(',');

  const results: Person[] = [];

  for (let i = 1; i < lines.length; i++) {
    const values = lines[i].split(',');
    if (values.length !== headers.length) {
      continue;
    }

    const nameStr = values[headers.indexOf('Name')].trim();
    const birthdayStr = values[headers.indexOf('Birthday')].trim();
    const diedStr = values[headers.indexOf('Died')].trim();
    const fatherStr = values[headers.indexOf('Father')].trim();
    const motherStr = values[headers.indexOf('Mother')].trim();
    const brotherStr = values[headers.indexOf('Brother')].trim();
    const sisterStr = values[headers.indexOf('Sister')].trim();

    const { firstName, lastName } = parseName(nameStr);
    const age = calculateAge(birthdayStr);

    const relatives: Relative[] = [];

    if (fatherStr && fatherStr !== 'null') {
      const { firstName: fFirstName, lastName: fLastName } = parseName(fatherStr);
      relatives.push({ FirstName: fFirstName, LastName: fLastName, Relationship: 'Father' });
    }

    if (motherStr && motherStr !== 'null') {
      const { firstName: mFirstName, lastName: mLastName } = parseName(motherStr);
      relatives.push({ FirstName: mFirstName, LastName: mLastName, Relationship: 'Mother' });
    }

    if (brotherStr && brotherStr !== 'null') {
      const { firstName: bFirstName, lastName: bLastName } = parseName(brotherStr);
      relatives.push({ FirstName: bFirstName, LastName: bLastName, Relationship: 'Brother' });
    }

    if (sisterStr && sisterStr !== 'null') {
      const { firstName: sFirstName, lastName: sLastName } = parseName(sisterStr);
      relatives.push({ FirstName: sFirstName, LastName: sLastName, Relationship: 'Sister' });
    }

    const person: Person = {
      FirstName: firstName,
      LastName: lastName,
      Birthday: formatDate(parseBirthday(birthdayStr)),
      Age: age,
      Relatives: relatives
    };

    results.push(person);
  }

  console.log(JSON.stringify(results, null, 2));
}

main();